node/network/availability-recovery/src/lib.rs

```diff
@@ -285,7 +290,11 @@ impl RequestChunksPhase {
 
 	async fn wait_for_chunks(&mut self, params: &InteractionParams) {
 		// Wait for all current requests to conclude or time-out, or until we reach enough chunks.
-		while let Some(request_result) = self.requesting_chunks.next().await {
+		// We will also stop, if there has not been a response for `MAX_CHUNK_WAIT`, so
+		// `launch_parallel_requests` cann fill up slots again.
+		while let Some(request_result) =
+			self.requesting_chunks.next().timeout(MAX_CHUNK_WAIT).await.flatten()
```

Comment:
Previously, a timeout would lead to an invocation of `self.shuffling.push_front(validator_index)`. Now, it no longer does. This could also lead to mistaken "unavailable" returns from the process.

Reply:
It still does. I did not change that code. The above timeout is a soft timeout, we still wait for the futures to return and they will still return with a timeout, triggering the `push_front` call. The only difference is that in the meantime we filled up free slots with new futures.

Reply:
Ah, I see. The timeout causes the loop to break so we do another loop iteration where we populate chunks.